### PR TITLE
Storage Group FT Fixs

### PR DESCRIPTION
**Pull request by doriac11**
# Description
- While writing the automation for the functional testing of the Storage Group datasource and resource. These issues were found and fixed.

# ISSUE TYPE
- Bugfix Pull Request

##### RESOURCE OR DATASOURCE NAME
StorageGroup

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [ ] Acceptance tests

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

**Comment by doriac11** [comment]
run e2e test

**Comment by doriac11** [comment]
run e2e test

File: examples/data-sources/powermax_storagegroup/data-source.tf [hunk 1]
```diff
@@ -16,7 +16,7 @@ limitations under the License.
 */
 data "powermax_storagegroup" "test" {
   filter {
-    names = ["esa_sg572"]
+    names = ["example_sg"]
```
Comment (nikitajoshi1) [inline examples/data-sources/powermax_storagegroup/data-source.tf:19]: @doriac11 Please check if UTs and FTs are working correctly after these changes.

**Review by nikitajoshi1 — APPROVED** [review]

File: examples/data-sources/powermax_storagegroup/data-source.tf (hunk 1 above)
Comment (doriac11) [reply examples/data-sources/powermax_storagegroup/data-source.tf:19]: They are @nikitajoshi1 , this is only in the example file, not the test itself.

**Comment by doriac11** [comment]
run e2e test

File: powermax/provider/storagegroup_resource.go [hunk 2]
```diff
@@ -72,6 +75,14 @@ func (r *StorageGroup) Schema(ctx context.Context, req resource.SchemaRequest, r
 				Required:            true,
 				Description:         "The name of the storage group",
```
Comment (shekhar-j) [inline powermax/provider/storagegroup_resource.go:76]: @doriac11 can you update the description and so related documentation generation?

**Review by shekhar-j — APPROVED** [review]

File: powermax/provider/storagegroup_resource.go (hunk 2 above)
Comment (doriac11) [reply powermax/provider/storagegroup_resource.go:76]: Yes will do!

Comment (doriac11) [reply powermax/provider/storagegroup_resource.go]: Done

**Pull request merged**